src/topography.f90

```diff
@@ -247,46 +248,42 @@ subroutine topography_deseas(this)
 
     do its = 1, 150   ! Only need high number after massive editing session with fjords. Normally 10 or so sweeps works.
       counter = 0
-      sea_num = 1
+      sea_num = 0
 
       ! Get number of seas
       do j = 2, this%nyt - 1
         i = 1
-        jm = j - 1
-        jp = j + 1
         if (sea(i, j) < land  .and. sea(i, j) > 0) then
-          if (sea(i, j) >= sea_num) then
-            sea(i, j) = sea_num
+          if (sea(i, j) > sea_num) then
             sea_num = max(min(sea_num+1, sea(this%nxt, j), sea(i, j-1), sea(i+1, j), sea(i, j+1)), sea_num)
+            sea(i, j) = sea_num
           end if
         end if
 
         do i = 2, this%nxt - 1
-          im = i - 1
-          ip = i + 1
           if (sea(i, j) < land  .and. sea(i, j) > 0) then
-            if (sea(i, j) >= sea_num) then
-              sea(i, j) = sea_num
+            if (sea(i, j) > sea_num) then
               sea_num = max(min(sea_num+1, sea(i-1, j), sea(i, j-1), sea(i+1, j), sea(i, j+1)), sea_num)
+              sea(i, j) = sea_num
             end if
           end if
         end do
 
         i = this%nxt
         if (sea(i, j) < land  .and. sea(i, j) > 0) then
-          if (sea(i, j) >= sea_num) then
-            sea(i, j) = sea_num
+          if (sea(i, j) > sea_num) then
             sea_num = max(min(sea_num+1, sea(i-1, j), sea(i, j-1), sea(1, j), sea(i, j+1)), sea_num)
+            sea(i, j) = sea_num
           end if
         end if
       end do
 
       j = this%nyt
       do i = 2, this%nxt - 1
```

Comment:
what about i=1, i=this%nxt when j = this%nyt?
(Not important in practice, since they are on land with a tripolar grid...)

Reply:
See comment bellow.